lib/github_s.mli

```diff
@@ -882,6 +882,12 @@ module type Github = sig
         {{:https://docs.github.com/rest/reference/git#references}git reference}
         with name [name] for repo [user]/[repo]. *)
 
+    val get_commits :
+      ?token:Token.t ->
+      user:string -> repo:string ->
+      unit -> Github_t.commits Response.t Monad.t
+    (** [get_commits ~user ~repo ()] lists all commits in [user]/[repo]. *)
```

Comment:
This could be improved by using `Stream.t` since the response is paginated, the Stream module provides an abstraction over GitHub's paginated endpoints. 
```suggestion
    val get_commits :
      ?token:Token.t ->
      user:string -> repo:string ->
      unit -> Github_t.commit Stream.t
    (** [get_commits ~user ~repo ()] stream of all commits in [user]/[repo]. *)
```